src/HeaderCell.tsx

```diff
@@ -87,7 +87,7 @@ export default function HeaderCell<R, SR>({
       const { right, left } = currentTarget.getBoundingClientRect();
       const width = isRtl ? right + offset - event.clientX : event.clientX + offset - left;
       if (width > 0) {
-        onColumnResize(column, width);
+        onColumnResize(column, clampColumnWidth(width, column));
```

Comment:
Otherwise `DataGrid.onColumnResize` is called with wrong width as it does not respect min/max width